### PR TITLE
[SYCLomatic] Add bfloat16 tests

### PR DESCRIPTION
**Pull request by 0x12CC**
Signed-off-by: Michael Aziz <michael.aziz@intel.com>

### CODE REVIEW

File: features/feature_case/math/math-bfloat16.cu [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// ===----------------------------------------------------------------------===//
+
+#include <cuda.h>
+#include <cuda_bf16.h>
+#include <iostream>
+
+__global__ void testMathFunctions(char *const TestResults) {
+  const auto bf16 = __float2bfloat16(3.14f);
+  const float f32 = __bfloat162float(bf16);
+  TestResults[0] = (fabs(3.14f - f32) < 0.1f);
```
Comment (zhimingwang36) [inline features/feature_case/math/math-bfloat16.cu]: how about we just comare the result of bf16 and f32 with CUDA result directly?

File: features/feature_case/math/math-bfloat16.cu [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#include <cuda_bf16.h>
+#include <iostream>
+
+__device__ bool valuesAreClose(float a, float b, float epsilon = 0.05f) {
+  return (fabs(a - b) < epsilon);
+}
+
+__global__ void testMathFunctions(char *const TestResults) {
+  const auto bf16 = __float2bfloat16(3.14f);
+  const float f32 = __bfloat162float(bf16);
```
Comment (zhimingwang36) [inline features/feature_case/math/math-bfloat16.cu:29]: Please add comment to tell the reason on:  why here not check exactly same data, but check the data is very close.

Comment (0x12CC) [reply features/feature_case/math/math-bfloat16.cu:29]: I've updated the test to include a comment.

File: features/feature_case/math/math-bfloat16.cu (hunk 1 above)
Comment (0x12CC) [reply features/feature_case/math/math-bfloat16.cu]: I've added a comparison to the bf16 CUDA results.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**